python/lsst/multiprofit/fitting/fit_psf.py

```diff
@@ -0,0 +1,639 @@
+            component_group_fluxes = [
+                component_group.get_fluxes_default(
+                    channels=channels,
+                    component_configs=component_group.get_component_configs(),
+                    is_fractional=component_group.is_fractional,
+                )
+                for component_group in self.model.component_groups.values()
+            ]
+
+        psf_model, priors = self.model.make_psf_model(component_group_fluxes=component_group_fluxes)
+        return psf_model
+
+    def schema_configurable(self) -> list[ColumnInfo]:
+        columns = []
+        if self.config_fit.eval_residual:
+            columns.append(ColumnInfo(key="n_eval_jac", dtype="i4"))
+        return columns
+
+    def schema(
+        self,
+        bands: list[str] = None,
+    ) -> list[ColumnInfo]:
+        """Return the schema as an ordered list of columns.
+
+        Parameters
+        ----------
+        bands
+            The bands to add band-dependent columns for.
+        """
+        if bands is not None:
+            if len(bands) != 1:
+                raise ValueError("CatalogPsfFitter must have exactly one band")
+        schema = super().schema(bands)
+        parameters = CatalogPsfFitterConfigData(config=self).parameters
+        unit_size = "pix"
+        units = {
+            g2f.ReffXParameterD: unit_size,
+            g2f.ReffYParameterD: unit_size,
+            g2f.SizeXParameterD: unit_size,
+            g2f.SizeYParameterD: unit_size,
+        }
+        schema.extend(
+            [
+                ColumnInfo(key=key, dtype="f8", unit=units.get(type(param)))
+                for key, param in parameters.items()
+            ]
+        )
+        schema.extend(self.schema_configurable())
+
+        return schema
+
+    def setDefaults(self):
+        self.prefix_column = "mpf_psf_"
+        self.compute_errors = "NONE"
+
+
+class CatalogPsfFitterConfigData(pydantic.BaseModel):
+    """A PSF fit configuration that can initialize models and images thereof.
+
+    This class relies on cached properties being computed once, mostly shortly
+    after initialization. Therefore, it and the config field must be frozen to
+    ensure that the model remains unchanged.
+    """
+
+    model_config = frozen_arbitrary_allowed_config
+
+    config: CatalogPsfFitterConfig = pydantic.Field(title="A CatalogPsfFitterConfig to be frozen")
+
+    @pydantic.field_validator("config")
+    @classmethod
+    def validate_config(cls, v: CatalogPsfFitterConfig) -> CatalogPsfFitterConfig:
+        v.validate()
+        return v
+
+    @cached_property
+    def components(self) -> dict[str, ComponentGroupConfig]:
+        components = self.psf_model.components
+        names = self.component_configs.keys()
+        if len(components) != len(names):
+            raise RuntimeError(f"{len(components)=} != {len(names)=}")
+        components_names = {name: component for name, component in zip(names, components)}
```

Comment:
It would be a pretty major bug if they didn't align since `self.psf_model` is populated by `self.config.make_psf_model` (hence the `RuntimeError`, which ought to be a `LogicError` - the only other way this could trigger is if a user somehow managed to get multiple components with the same name). I'm not sure if checking names is the best way to validate here but I will consider adding names to components/sources on DM-38617.